In [1]:
import pandas as pd
import numpy as np
import torch
from torch.utils.data import Dataset, DataLoader
from sklearn.preprocessing import OneHotEncoder
from skorch import NeuralNetClassifier

import sys  
sys.path.insert(0, '/root/Project-CS224N-ED-Disposition/')
from scripts import utils
import importlib
importlib.reload(utils)

index_col = 'Unnamed: 0'

In [4]:
ed = pd.read_csv('/root/data/ed.csv')
ed

index           CSN  Rads_order_ID  \
0            0  131290418105      674819827   
1            1  131290415760      674821169   
2            2  131290418527      674823260   
3            3  131290418527      674823261   
4            4  131290415760      674823291   
...        ...           ...            ...   
146144  146144  131336973092      813941915   
146145  146145  131336952819      813964164   
146146  146146  131336954468      814069582   
146147  146147  131336896882      814099628   
146148  146148  131336896882      814108675   

                                                    Study  \
0                                        XR CHEST 2 VIEWS   
1                                  CT HEAD WO IV CONTRAST   
2                         CT ABDOMEN PELVIS W IV CONTRAST   
3                                        XR CHEST 2 VIEWS   
4                           XR PELVIS AND HIP 1 VIEW LEFT   
...                                                   ...   
146144                     MR LUMBAR SPINE WO IV CONTRAST   
146145                    CT ABDOMEN PELVIS W IV CONTRAST   
146146  US LOWER EXTREMITY VEINS DEEP VEIN THROMBOSIS ...   
146147                                    XR CHEST 1 VIEW   
146148              CT CHEST ABDOMEN PELVIS W IV CONTRAST   

                                               Impression  \
0       1.  Interval increase in perihilar opacities a...   
1       1.  No acute intracranial abnormality. 2.  Mil...   
2       1.  Multiple loops of dilated small bowel with...   
3       1.  Interval increased loculated moderate meta...   
4       1.  Redemonstrated fracture of the femoral hea...   
...                                                   ...   
146144  1.  Acute superior endplate compression deform...   
146145  1.  Diffuse anasarca and mesenteric edema. Pat...   
146146  1.  No ultrasound evidence of deep venous thro...   
146147  1.  Mild fluid overload with new small pleural...   
146148  1.  Distended gallbladder with mild wall thick...   

                    ED_dispo  
0       Place in Observation  
1         Admit to Inpatient  
2         Admit to Inpatient  
3         Admit to Inpatient  
4         Admit to Inpatient  
...                      ...  
146144    Admit to Inpatient  
146145    Admit to Inpatient  
146146    Admit to Inpatient  
146147    Admit to Inpatient  
146148    Admit to Inpatient  

[146149 rows x 6 columns]

In [5]:
ed['ED_dispo'].value_counts()

Discharge                   71379
Admit to Inpatient          66309
Place in Observation         7530
Place in Observation-CDU      931
Name: ED_dispo, dtype: int64

In [6]:
71379 / 146149

0.4883988258558047

zero rate baseline: predicting the mode (always discharge) results in 48.8% accuracy

In [ ]:
TODO: try some simple classifiers like nearest neighbors and random forest